File: go/grains/grains.go
package grains

import "errors"

func Square(n int) (uint64, error) {
	if n < 1 || n > 64 {
		return 0, errors.New("n out of range 1..64")
	}
	return 1 << (n - 1), nil
}

func Total() uint64 {
	return 0xFFFFFFFFFFFFFFFF
}

File: c/grains/src/grains.c
#include "grains.h"

uint64_t square(uint8_t index) {
    return (0 < index && index <= 64) ? (uint64_t) 1 << (index - 1) : 0;
}

uint64_t total(void) {
    return 0xFFFFFFFFFFFFFFFF;
}